src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -230,7 +230,9 @@ public Task<TElement> DequeueAsync(CancellationToken cancellationToken = default
         /// Note: The early cancellation behavior is intentional.
         /// </summary>
         [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/23582", OftenCompletesSynchronously = true)]
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
         private static Task<T> WithCancellation<T>(Task<T> task, CancellationToken cancellationToken)
+#pragma warning restore VSTHRD200 // Use "Async" suffix for async methods
```

Comment:
Rather than suppress the check here I would rename the method. These are `private` APIs hence nothing really stopping us from changing them.